storage/grpc_client.go

```diff
@@ -1168,58 +1158,58 @@ func (c *grpcStorageClient) NewMultiRangeDownloader(ctx context.Context, params
 	// This is the size of the entire object, even if only a range was requested.
 	size := obj.GetSize()
 
-	rr := &gRPCBidiReader{
+	mrd := &gRPCBidiReader{
 		stream:           resp.stream,
 		cancel:           cancel,
 		settings:         s,
 		readHandle:       msg.GetReadHandle().GetHandle(),
 		readID:           1,
 		reopen:           openStream,
-		readSpec:         r,
-		data:             make(chan []rangeSpec, 100),
+		readSpec:         bidiObject,
+		rangesToRead:     make(chan []mrdRange, 100),
 		ctx:              ctx,
 		closeReceiver:    make(chan bool, 10),
-		closeManager:     make(chan bool, 10),
-		managerRetry:     make(chan bool), // create unbuffered channel for closing the streamManager goroutine.
+		closeSender:      make(chan bool, 10),
+		senderRetry:      make(chan bool), // create unbuffered channel for closing the streamManager goroutine.
 		receiverRetry:    make(chan bool), // create unbuffered channel for closing the streamReceiver goroutine.
-		mp:               make(map[int64]rangeSpec),
+		activeRanges:     make(map[int64]mrdRange),
 		done:             false,
-		activeTask:       0,
+		numActiveRanges:  0,
 		streamRecreation: false,
 	}
 
-	// streamManager goroutine runs in background where we send message to gcs and process response.
-	streamManager := func() {
-		var currentSpec []rangeSpec
+	// sender receives ranges from user adds and requests these ranges from GCS.
+	sender := func() {
+		var currentSpec []mrdRange
 		for {
 			select {
-			case <-rr.ctx.Done():
-				rr.mu.Lock()
-				rr.done = true
-				rr.mu.Unlock()
+			case <-mrd.ctx.Done():
+				mrd.mu.Lock()
+				mrd.done = true
+				mrd.mu.Unlock()
 				return
-			case <-rr.managerRetry:
+			case <-mrd.senderRetry:
 				return
-			case <-rr.closeManager:
-				rr.mu.Lock()
-				if len(rr.mp) != 0 {
-					for key := range rr.mp {
-						rr.mp[key].callback(rr.mp[key].offset, rr.mp[key].totalBytesWritten, fmt.Errorf("stream closed early"))
-						delete(rr.mp, key)
+			case <-mrd.closeSender:
+				mrd.mu.Lock()
+				if len(mrd.activeRanges) != 0 {
+					for key := range mrd.activeRanges {
+						mrd.activeRanges[key].callback(mrd.activeRanges[key].offset, mrd.activeRanges[key].totalBytesWritten, fmt.Errorf("stream closed early"))
```

Comment:
this could also be an errors.New I guess

Reply:
Agreed